### ISSUE TITLE
Unequal amount of rests for 2 voices

### ISSUE THREAD

**Issue by BestBacon123023** [musescore/MuseScore#22574]
### Issue type

UX/Interaction bug (incorrect behaviour)

### Bug description

I accidentally deleted a few rests on voice 2 while I was writing for 2 voices on the same stave, and it seens like I can’t input any more notes/rests past the ones that are not deleted in said voice.

### Steps to reproduce

1. Input rests on voice 1 and voice 2. (preferrably of the same length and amount, I did this with 8 8th rests in a 4/4 bar)
2. Delete a few rests from 1 of the voices.

### Screenshots/Screen recordings

![IMG_0909](https://github.com/musescore/MuseScore/assets/168100669/80643557-396d-454a-9ad8-7c1324220609)
![IMG_0910](https://github.com/musescore/MuseScore/assets/168100669/2bb218a6-555a-413f-a532-909fab62dd6b)

### MuseScore Version

4

### Regression

No.

### Operating system

MacOS 14

**Comment by BestBacon123023** [comment]
Note: You can’t input past the rests/notes that didn’t get deleted.
![IMG_0910](https://github.com/musescore/MuseScore/assets/168100669/4accce65-7909-4b87-878f-27b4d586d29f)

**Comment by BestBacon123023** [comment]
i accidentally closed this bug report

**Comment by BestBacon123023** [comment]
reopening this issue

**Comment by BestBacon123023** [comment]
hi, can someone help me reopen this issue?

**Comment by MarcSabatella** [comment]
Indeed, deleting rests has been problematic since the beginning.  There had been some discussion about simply disallowing that, as it leaves these uneditable holes, and having the delete command hide the rests instead.  Fixing the input system to fill those holes is possible but perhaps not trivial.

Meanwhile, I do suggest hiding instead of deleting rests to avoid this, but if you do end up with holes, one smile fix is to exchange voices 1-2 twice (via Edit / Voices).  Moving voice 2 to 1 automatically fills the holes.  In many cases you can also either delete the remaining notes & rests, or lengthen the last one to fill the measure, then shorten it again.

**Comment by BestBacon123023** [comment]
Thanks for the help!
